app/controllers/user_controller.rb

```diff
@@ -20,30 +20,30 @@ def signup
   def update_user
     current_user = User.find(params[:id])
     payload = JSON.parse(request.body.read)
-    current_user.first_name = payload["first_name"] unless payload['first_name']&.empty?
-    current_user.email = payload["email"] unless payload['email']&.empty?
-    current_user.last_name = payload["last_name"] unless payload['last_name']&.empty?
-    current_user.user_type = payload["user_type"] unless payload['user_type']&.empty?
-
-    if current_user.valid?
-      current_user.save
+    update_detail = custom_compact(payload)
+    begin
+      current_user.update!(update_detail)
```

Comment:
You could use `assign_attributes` to update the in-memory record without triggering a validation exception.

```rb
current_user.assign_attributes(update_detail)

if current_user.valid?
   # ...
end
```